repos/spack_repo/builtin/packages/darshan_runtime/package.py

```diff
@@ -104,9 +105,26 @@ class DarshanRuntime(AutotoolsPackage):
         default="none",
         description="Path to centralized, formatted Darshan log directory",
     )
-    variant("mmap_logs", default=False, description="Use mmap to store Darshan log data")
+
+    variant(
+        "mmap_logs",
+        default=False,
+        description="Use mmap to store Darshan log data",
+        when="@:3.4.7",
+    )
+
+    variant(
+        "mmap_logs", default=True, description="Use mmap to store Darshan log data", when="@3.5.0:"
+    )
+
     variant("group_readable_logs", default=False, description="Write group-readable logs")
 
+    def url_for_version(self, version):
+        if version < Version("3.5.0"):
+            return f"https://web.cels.anl.gov/projects/darshan/releases/darshan-{version}.tar.gz"
+        else:
+            return f"https://github.com/darshan-hpc/darshan/archive/refs/tags/{version}.tar.gz"
+
```

Comment:
```suggestion
    def url_for_version(self, version):
        if self.spec.satisfies("@:3.4"):
            return f"https://web.cels.anl.gov/projects/darshan/releases/darshan-{version}.tar.gz"
        else:
            return f"https://github.com/darshan-hpc/darshan/archive/refs/tags/{version}.tar.gz"

```

We can drop the additional version object creation here by instead using the `satisfies()` directive.